aws/usageReports/rds/monthlyReport.go

```diff
@@ -0,0 +1,146 @@
+//   Copyright 2018 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package rds
+
+import (
+	"time"
+	"strings"
+	"context"
+
+	"github.com/trackit/jsonlog"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/rds"
+
+	"github.com/trackit/trackit-server/config"
+	taws "github.com/trackit/trackit-server/aws"
+	"github.com/trackit/trackit-server/aws/usageReports"
+)
+
+// fetchMonthlyInstancesList fetches instances based on billing data
+func fetchMonthlyInstancesList(ctx context.Context, creds *credentials.Credentials, instList []utils.CostPerInstance,
+	region string, instanceChan chan Instance, startDate, endDate time.Time) error {
+	defer close(instanceChan)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(region),
+	}))
+	svc := rds.New(sess)
+	for _, inst := range instList {
+		desc := rds.DescribeDBInstancesInput{DBInstanceIdentifier: aws.String(inst.Instance)}
+		instances, err := svc.DescribeDBInstances(&desc)
+		if err != nil {
+			continue
```

Comment:
Why do you not return the error or at least log it ?

Reply:
An error is automaticly return by `DescribeDBInstances` if I try to get an instance on the wrong region (I try every instance with every region). So it's not really an error.

Reply:
How could you differentiate between an error that is in the wrong region, and an error with the RDS request describing the instances ?